### PR TITLE
test: UpdateMangaStatusAndMissingChapterCount test coverage

### PR DESCRIPTION
**Pull request by nonproto**
💡 What: Added comprehensive unit tests for `UpdateMangaStatusAndMissingChapterCount` covering sad paths (e.g., database exceptions), missing chapters string calculation, and completion status updates. Added `kotlinx-coroutines-test` dependency via version catalog to correctly implement the tests utilizing `runTest`. Reverted unrelated formatting changes.

🎯 Why: To ensure edge cases and specific flow logics are tested, making this previously untested business logic robust and safe from regressions while keeping PR scoped correctly.

---
*PR created automatically by Jules for task [15261991602516363197](https://jules.google.com/task/15261991602516363197) started by @nonproto*

### CODE REVIEW

File: app/src/test/java/org/nekomanga/usecases/manga/UpdateMangaStatusAndMissingChapterCountTest.kt [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+            coEvery { insertMangaMock.executeOnIO() } returns mockk()
+
+            // Act
+            updateMangaStatusAndMissingChapterCount(manga)
+
+            // Assert
+            assertEquals(SManga.COMPLETED, manga.status)
+            coVerify(exactly = 1) { db.insertManga(manga) }
+        }
+}
```
[the remark this answers is not in the document]
Comment (nonproto) [reply app/src/test/java/org/nekomanga/usecases/manga/UpdateMangaStatusAndMissingChapterCountTest.kt:218]: > high
> 
> The tests in this class, particularly `given valid chapters list with missing chapters...` and `given manga cancelled and final chapter available...`, are heavily dependent on mocking static extension functions. These mocks are enabled globally in the `setup` method, which leads to tests that don't actually verify the core business logic they are named after. For example, mocking the result of `getMissingChapters` means the test doesn't validate the chapter calculation logic at all.
> 
> This approach makes the tests brittle and less valuable, as they are testing the mock setup rather than the behavior of the use case.
> 
> To improve this, I recommend the following:
> 
> 1. **Reduce global mocks**: Move the `mockkStatic` calls from the `setup` method into the specific tests that absolutely require them (like the exception handling test).
> 2. **Test the behavior, not the implementation**: For the tests covering missing chapters and status updates, you should create real `ChapterImpl` objects with the necessary properties and let the underlying functions (`getMissingChapters`, `isAvailable`, etc.) execute. This will create more meaningful integration tests for the use case.
> 
> The test for the exception path (`given database throws exception...`) is a good example of appropriate mocking. The other tests should be refactored to truly test the business logic.

@jules
